spec/person_spec.rb

```diff
@@ -0,0 +1,12 @@
+require_relative '../person'
+
+describe 'Person' do
+  before(:each) do
+    @person = Person.new('55', 'roshan', parent_permission: true)
+  end
+  it 'Take age, name and parent_permission' do
+    expect(@person.age).to eq '55'
+    expect(@person.name).to eq 'roshan'
+    expect(@person.parent_permission).to eq true
+  end
```

Comment:
- [ ] Hello team, great job so far, but since are are aiming for 100% test coverage, we still have have some work to do. We need to cover all the methods in all the classes with tests. That said, for the `person class` for example, we can add the following tests;
- constructor method sets default value for the optional name parameter if it is not provided like so:
```ruby
    it "The constructor method sets default value for the optional name parameter if it is not provided" do
      person = Person.new(age)
      expect(person.name).to eql "Unknown"
    end
```
- adding a new rental like so;
```ruby
  context "When adding a new rentals" do

    it "The add_rental method returns a rental and adds its to the book" do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Bob")

      rental = person.add_rental(book, "2021-05-04")

      expect(person.rentals).to eql [rental]
    end
  end
```
- checking for permission
```ruby
  context "When checking permissions" do
    
    it "The can_use_services? method returns false if under age and no parent permission" do
      person = Person.new(14, "Rose", false)
      expect(person.can_use_services?).to be false
    end
    
    it "The can_use_services? method returns true if over age" do
      person = Person.new(19, "Sam", false)
      expect(person.can_use_services?).to be true
    end
    
    it "The can_use_services? method returns true if has parent permission" do
      person = Person.new(14, "Rose", true)
      expect(person.can_use_services?).to be true
    end
  end
```
- validating name 
```ruby
  context "When validating name" do

    it "The validate_name method corrects the name" do
      person = Person.new(19, "alongnameiscorrected")
      person.validate_name()
      expect(person.name).to eql "Alongnamei"
    end
  end
```
### The above code snipets are just examples, let's make sure all the methods in all our classes are  covered with tests.